util.sh

```diff
@@ -0,0 +1,147 @@
+# ============================================================================
+# DOCKER
+# ============================================================================
+
+# wrapper around docker-compose except:
+# * in CI it switches to using images for caching
+# * locally uses docker-compose as-is which builds image from scratch
+# this allows to easily update containers as necessary locally
+# without needing to build+tag images
+function compose {
+    if [ -n "${CI:-}" ]; then
+        # sed uncomments lines starting with "# CI"
+        # and removes lines ending with "# CI"
+        # therefore allowing to switch what is used locally and in CI
+        set -x
+        exec docker-compose \
+            -f <(cat docker-compose.yml | sed -r -e 's/^(\s+)# CI\s*(.*)/\1\2/g' -e '/# CI/d') \
+            --project-directory $PWD \
+            $@
+    else
+        set -x
+        exec docker-compose $@
+    fi
+}
+
+# if Makefile target specifies which docker-compose service to use
+# get its name to be used with docker-compose
+# otherwise fallback to provided default
+# usage:
+# service_for_compose <default> "$@"
+function service_for_compose {
+    default=$1
+    args=$2
+    # looks magical but its pretty simple
+    # for example if script is called with ./build.sh lint
+    # it will look for a line in Makefile starting with "lint: # docker-compose:"
+    # and if it finds such a line, it extracts docker-compose service
+    # which is expected to be used for that makefile target
+    service=$(
+        (
+            grep -E "^($(
+                echo $args \
+                    | tr ' ' '\n' \
+                    | paste -s -d'|'
+            )): # docker-compose:" Makefile \
+                || true
+        ) \
+            | cut -d: -f3 \
+            | head -n1
+    )
+    echo ${service:-$default}
+}
+
+# ============================================================================
+# HELP
+# ============================================================================
+
+ORANGE="\033[0;33m"
```

Comment:
nit: this is yellow, not orange

Reply:
haha. copy-paste across repos for a while was using this as orange

Reply:
adjusted